google-cloud-bigquerystorage/src/test/java/com/google/cloud/bigquery/storage/v1beta2/CivilTimeEncoderTest.java

```diff
@@ -0,0 +1,425 @@
+/*
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.bigquery.storage.v1beta2;
+
+import static org.junit.Assert.assertEquals;
+
+import java.util.logging.Logger;
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import org.threeten.bp.LocalDateTime;
+import org.threeten.bp.LocalTime;
+
+@RunWith(JUnit4.class)
+public class CivilTimeEncoderTest {
+  private static final Logger LOG = Logger.getLogger(CivilTimeEncoderTest.class.getName());
+
+  @Test
+  public void Packed64TimeMicrosTest() {
+    // 00:00:00.000000
+    // 0b000000000000000000000000000|00000|000000|000000|00000000000000000000
+    // 0x0
+    assertEquals(0x0L, CivilTimeEncoder.encodePacked64TimeMicros(LocalTime.of(0, 0, 0, 0)));
```

Comment:
Is there a case for if user pass in nannos?

Reply:
I've added a simple case where passing in 1 nanosecond should return 0 since it won't reach a whole microsecond. I have other tests where I pass in millions of nanos to get microseconds, is there anything else I need to cover?

Reply:
Should i do the same for DateTimeSeconds and DateTimeMicros?

Reply:
Okay I've added a test case and the current behavior is that if a user passes in a nonsecond that can be directly converted to a microsecond, the conversions are fine, if the nanos do not convert and precision will be lost, the conversion will throw an error.